### ISSUE TITLE
cactus fails to create pull request for root

### ISSUE THREAD

**Issue by jonathanl-telenav** [Telenav/cactus-build#49]
Creating a pull request for feature/testing results in the following output. Even though it says it put telenav-build in the pull request set, it thinks there is no PR.

```
[INFO] --- cactus-maven-plugin:1.5.24:git-pull-request (default-cli) @ telenav-build ---
[INFO] Ignoring matched checkout cactus for pull request - because we are matching the branch feature/testing but it is on the branch develop

[SNIP]

[INFO] Ignoring matched checkout telenav-superpom for pull request - because we are matching the branch feature/testing but it is on the branch develop
[INFO] Will include telenav-build in the pull request set, on branch feature/testing
[WARNING] Every checkout matched already has an open, mergeable PR.  Nothing to do.

**Comment by timboudreau** [comment]
Is there actually a branch called `feature/testing` of `telenav-build`?

If yes, that's a bug.  If no, it's as intended.

**Comment by jonathanl-telenav** [comment]
I created it. I only pushed it because of the problem.

**Comment by timboudreau** [comment]
You created it *after* encountering the problem, or it already existed when the create pull request mojo was invoked?

How was the create pull request mojo invoked?

**Comment by timboudreau** [comment]
Was your checkout of telenav-build **on that branch** at the time?

**Comment by timboudreau** [comment]
Okay, this is really an ordering issue, that the mojo worked around by deciding that if they only checkout that needed a branch change was the root, then it has nothing to do.

What the mojo needs to do is to ensure that the submodule root (or really, any submodule root - there could be multiple, though we don't have them) is processed **first**.  Otherwise, a naive fix results in the submodules getting moved to the target branch first, which *generates changes in the submodule root, putting it into a modified state* - and that, then means that git refuses to change branches because the submodule root checkout is dirty - because all of the heads of the submodules were changed to the head of the feature branch.

A secondary problem is that we use `git diff --quiet` to determine if there are local modifications to commit, but in the submodule root, that will exit with 1 even if the state is `modified:   cactus (modified content)`.  We need to actually detect that situation - otherwise we will try to generate a new commit in the root (which is the right thing if we modified `.gitmodules` or actually created a new branch) - but will fail when it tries to run `git add -A` and `git commit -m` to generate a new commit in the root, because `git add -A` doesn't add anything, so the commit will fail.
